web/cobrands/fixmystreet/assets.js

```diff
@@ -227,10 +241,14 @@ function asset_selected(e) {
 
 function asset_unselected(e) {
     fixmystreet.markers.setVisibility(true);
+    selected_feature = null;
+    if (e.feature.attributes._preserve_form_values_on_unselect) {
```

Comment:
Looks like this works fine, and thanks for the big comment further up :) Just wondering why you need to set this value at all, though, can't you do `if (this.fixmystreet.disable_pin_snapping)` in `asset_unselected` as well?